src/behaviors/Selectable/Selectable.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { noop } from 'lodash';
+
+class Selectable extends React.Component {
+  static propTypes = {
+    /**
+     * Render prop to render the selectable items. Supports the following props:
+     *
+     * *selectItem*: A function that should be invoked with a unique key as its argument whenever an item should be selected.
+     *
+     * *deselectItem*: A function that should be invoked with a unique key as its argument whenever an item should be deselected.
+     *
+     * *toggleItem*: A function that should be invoked with a unique key as its argument whenever an item should be toggled.
+     *
+     * *selected*: A **Set** of selected items represented by their key values.
+     */
+    render: PropTypes.func,
+  };
+  state = { selected: new Set() };
+
+  static defaultProps = {
+    render: noop,
+  };
+
+  selectItem = key => {
+    const { state: { selected } } = this;
+    selected.add(key);
+    this.setState({ selected });
```

Comment:
This should be a function-based state update to avoid race conditions. Playing around with this locally, I was able to click items fast enough that the state could not keep up.

Reply:
I'm not really sure what you mean by function-based state update.

Reply:
You can call `this.setState(currentState => ({ selected: currentState.selected.add(key) });` and it will enqueue that modification as a discrete one on the internal buffer, so it will avoid conflicting with other concurrent state changes.

I don't think the built-in `Set`'s `.add` returns the set, though... for that reason I usually prefer to import `immutable`'s `Set`.

Reply:
`Set.add()` does but `Set.delete()` does not.